packages/ciscospark/test/integration/spec/plugins/whiteboards.js

```diff
@@ -0,0 +1,127 @@
+import {base64} from '@ciscospark/common';
+
+describe(`ciscospark`, function() {
+  this.timeout(60000);
+
+  describe(`#whiteboards`, () => {
+
+    beforeEach(() => spark.device.register());
+    afterEach(() => spark.device.unregister());
+
+    const rooms = [];
+    afterEach(() => {
+      return Promise.all(rooms.map((room) => {
+        return spark.rooms.remove(room)
+          .catch((reason) => {
+            console.error(`Failed to delete room`, reason);
+          });
+      }))
+        .then(() => {
+          while (rooms.length) {
+            rooms.pop();
+          }
+        });
+    });
+
+    describe(`#create()`, () => {
+      it(`creates a whiteboard for a room`, () =>
+        spark.rooms.create({title: `Cisco Spark Test Room`})
+          .then((room) => {
+            rooms.push(room);
+            const body = {roomId: room.id};
+            return spark.whiteboards.create(body);
+          })
+          .then((board) => {
+              assert.isWhiteboard(board);
+              assert.isHydraID(board.roomId);
+              return spark.whiteboards.get(board.id)
+                .then((got) => {
+                  assert.isWhiteboard(got);
+                  assert.equal(got.id, board.id);
+                  assert.equal(got.roomId, board.roomId);
+                });
+          })
+      );
+    });
+
+
+    describe(`#get()`, () => {
+      it(`get a nonexistent whiteboard throws not found`, () =>
+        spark.whiteboards.get('Y2lzY29zcGFyazovL3VzL1RFQU0vMTNlMThmNDAtNDJmYy0xMWU2LWE5ZDgtMjExYTBkYzc5NzY5')
+          .then((got) => {
+            assert.fail(got, null, 'Expected error');
+          })
+          .catch((err) => {
+            assert.statusCode(err, 404, 'expected Not Found');
+          })
+      );
+
+      it(`supports get by channel uuid`, () =>
+        spark.rooms.create({title: `Cisco Spark Test Room`})
+          .then((room) => {
+            rooms.push(room);
+            const body = {roomId: room.id};
+            return spark.whiteboards.create(body);
+          })
+          .then((board) => {
+            let boardId = base64.decode(board.id);
+            const idx = boardId.lastIndexOf('/');
+            boardId = boardId.substring(idx + 1);
+            return spark.whiteboards.get(boardId)
+              .then((got) => {
+                assert.isWhiteboard(got);
+                assert.equal(got.id, board.id);
+                assert.equal(got.roomId, board.roomId);
+              });
+          })
+      );
+    });
+
+    describe(`#list()`, () => {
+      let boards;
+      const numRooms = 2;
+      const numBoards = 2;
+      beforeEach(() => {
+        boards = {};
+        return createRooms(numRooms, spark)
+          .then((roomList) => Promise.all(roomList.map((room) => {
+            rooms.push(room);
+            boards[room.id] = [];
+            return createBoards(numBoards, room.id, spark)
+              .then((boardList) =>
+                Promise.all(
```

Comment:
I try to keep implicit returns on the same line as their arrows. in case where that would put two arrows on the same line, I replace one of the implicit returns with an explicit return and use `// eslint-disable-next-line` to get past the lint issue.